valet-key/ValetKey.Web/Controllers/HomeController.cs

```diff
@@ -52,36 +56,33 @@ public ActionResult Index()
 
         private async Task<StorageEntitySas> GetSharedAccessReferenceForDownload(string blobName)
         {
-            var blobClient = this.account.CreateCloudBlobClient();
-            var container = blobClient.GetContainerReference(this.blobContainer);
-
-            var blob = container.GetBlockBlobReference(blobName);
-            
-            if (!await blob.ExistsAsync())
-            {
-                throw new Exception("Blob does not exist");
-            }
+            var container = blobServiceClient.GetBlobContainerClient(this.blobContainer);
```

Comment:
Do you need a container client or can you create a blobClient directly using the container name and blob name?